Common/File/Path.cpp

```diff
@@ -198,11 +251,44 @@ bool Path::CanNavigateUp() const {
 }
 
 Path Path::NavigateUp() const {
+	if (type_ == PathType::CONTENT_URI) {
+		AndroidContentURI uri(path_);
+		uri.NavigateUp();
+		return Path(uri.ToString());
+	}
 	std::string dir = GetDirectory();
 	return Path(dir);
 }
 
+Path Path::GetRootVolume() const {
```

Comment:
Hm, this method name is slightly misleading given both Windows and Linux/macOS support mounting volumes at paths.  I guess it's fine and our current usage makes sense.

-[Unknown]

Reply:
I'll add a comment.